components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/listener/IdentityMgtEventListener.java

```diff
@@ -554,11 +558,24 @@ private void handleEvent(String userName, UserStoreManager userStoreManager, Str
             if (StringUtils.isNotBlank(roleName)) {
                 properties.put(IdentityEventConstants.EventProperty.ROLE_NAME, roleName);
             }
+
+            int tenantId = userStoreManager.getTenantId();
+            String userTenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            try {
+                RealmService realmService = IdentityMgtServiceDataHolder.getInstance().getRealmService();
+                TenantManager tenantManager = realmService.getTenantManager();
+                userTenantDomain = tenantManager.getDomain(tenantId);
+            } catch (Exception e) {
```

Comment:
Is it necessary to catch 'Exception' here?  Can't we catch any specific exceptions?